records.py

```diff
@@ -282,40 +296,42 @@ def get_table_names(self, internal=False):
         # Setup SQLAlchemy for Database inspection.
         return inspect(self._engine).get_table_names()
 
-    def get_connection(self):
+    def get_connection(self, close_with_result=False):
         """Get a connection to this Database. Connections are retrieved from a
-        pool.
+        pool. By default, the retrieved connection remains open. Setting
+        close_with_result to True returns the connection to the pool once a
+        result is consumed.
         """
         if not self.open:
             raise exc.ResourceClosedError('Database closed.')
 
-        return Connection(self._engine.connect())
+        return Connection(self._engine.contextual_connect(close_with_result))
```

Comment:
contextual_connect is deprecated:
https://docs.sqlalchemy.org/en/13/core/connections.html#sqlalchemy.engine.Engine.contextual_connect

I removed contextual_connect just to see what happens, and it turns out: Can be removed and tests still pass.

Reply:
Good catch. I also verified that all of the tests work with sqlite and postgres.